### DISCUSSION TITLE
Service not restarting

### DISCUSSION THREAD

**Discussion by calabresemic** [travisghansen/hass-pfsense#57] [Q&A]
Hey I've set up the pfsense integration in my home assistant instance and I can see all the entities reported by pfsense but I haven't been able to get my automation to trigger a restart of the service yet. Since there aren't many examples out there yet I was wondering what I'm doing wrong? I've tried both openvpn and openvpn-client for the service name, ideally I just want to restart the client when my VPN starts acting up but I can deal with it restarting all of openvpn if it comes to it. Any help is greatly appreciated.

`  alias: Restart VPN
  description: ''
  trigger:
  - platform: state
    entity_id: sensor.pfsense_gateway_privadovpn_vpnv4_status
    to: offline
  condition: []
  action:
  - service: pfsense.stop_service
    data:
      entity_id: binary_sensor.pfsense_pending_notices_present
      service_name: openvpn-client
  mode: single`

**Comment by travisghansen** [comment]
OpenVPN services are ‘special’ as I recall because there is a service per configuration in essence. I’d have to dig through the code to remember exactly what needs to be invoked to restart those but my guess is an openvpn specific service could be add to the integration that would be relatively easy to invoke.

**Reply by calabresemic** [reply]
That would be great. Let me know if you need anything from me to help make that happen.

**Reply by patrik996** [reply]
Im also trying to stop/start OpenVPN client. Would be great if it could be added as a switch.
